### PR TITLE
fastfetch 1.7.2 (new formula)

### PR DESCRIPTION
**Pull request by CarterLi**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Like neofetch, but much faster because written in C

I'm the one of the contributors who maintain the macOS part

### CODE REVIEW

File: Formula/fastfetch.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  url "https://github.com/LinusDierheimer/fastfetch/archive/refs/tags/1.7.2.tar.gz"
+  sha256 "f0778cdc6d33f5399bb82a8689b816ec3ff1d6b3eb74488e88d7a92a0bd3f7b6"
+  license "MIT"
+  head "https://github.com/LinusDierheimer/fastfetch.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "libplist" => :build
+  depends_on "pkg-config" => :build
+  depends_on "vulkan-loader" => :build
+  depends_on :macos
```
Comment (SMillerDev) [inline Formula/fastfetch.rb:13]: Why not Linux?

File: Formula/fastfetch.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  def install
+    system "cmake", "-S", ".", "-B", "build", "-DCMAKE_INSTALL_SYSCONFDIR=#{etc}", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  def caveats
+    <<~EOS
+      Install libplist for iTerm terminal font detection
+    EOS
```
Comment (SMillerDev) [inline Formula/fastfetch.rb:24]: This should be in upstream documentation, it doesn't seem related to homebrew.

Comment (CarterLi) [reply Formula/fastfetch.rb]: Removed

File: Formula/fastfetch.rb (hunk 1 above)
Comment (CarterLi) [reply Formula/fastfetch.rb]: I dont have a linux machine to test and according to the official document fastfetch requires a lot dependencies to be useful.

However fastfetch should build without any dependencies. I will try to make it work on linux

File: Formula/fastfetch.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  depends_on "vulkan-loader" => :build
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", "-DCMAKE_INSTALL_SYSCONFDIR=#{etc}", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    assert_match "fastfetch", shell_output("#{bin}/fastfetch --version")
```
Comment (chenrui333) [inline Formula/fastfetch.rb:20]:
```suggestion
    assert_match "fastfetch #{version}", shell_output("#{bin}/fastfetch --version")
```

Comment (CarterLi) [reply Formula/fastfetch.rb:20]: It doesn't work with `--HEAD`. I think the current code is enough to make sure that fastfetch doesn't crash at start.

Comment (chenrui333) [reply Formula/fastfetch.rb:20]: I see.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@CarterLi, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!
